crates/ruff_python_formatter/src/expression/mod.rs

```diff
@@ -199,6 +201,10 @@ impl Format<PyFormatContext<'_>> for MaybeParenthesizeExpression<'_> {
             OptionalParentheses::Always => {
                 expression.format().with_options(Parentheses::Always).fmt(f)
             }
+            OptionalParentheses::Never if *parenthesize == Parenthesize::IfBreaksOrIfRequired => {
```

Comment:
I think you still want an extra case for `Multiline` and `IfBreaksOrRequired` because parentheses are now omitted if `can_omit_optional_parentheses` returns `True`. Is is this what we want? Can we come up with a test case that expands over multiple lines and `can_omit_optional_parentheses` returns `True`.

Reply:
You're right, good call.